portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -34,8 +34,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String userEmail = userService.getCurrentUser().getEmail();
 
       String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+    
+      if (nickname == ""){ out.println("<p>You don't have a nickname</p>");
```

Comment:
A couple comments:

(1) for nickname == "", are you supposed to check emptiness? If so, I think you should use nickname.isEmpty(). This right now is comparing reference equality.

(2) Use a new line after {, and add a space between ){.